### DISCUSSION TITLE
Troubles with connecting encoder to stm32

### DISCUSSION THREAD

**Discussion by retroretro954** [davetcc/tcMenu#423] [Q&A]
stm32f103c8, stm32duino. Tried different pins, different encoders, tried with and without pullup logic. Tried connecting encoder as buttons. Maybe i'm missing something?

**Comment by retroretro954** [comment]
Tried like this: 
    switches.init(internalDigitalIo(), SWITCHES_POLL_EVERYTHING, true);
    menuMgr.initForEncoder(&renderer, &menuSensorTemp, PB_8, PB_10, PB_9);

**Reply by retroretro954** [reply]
compiler doesn't let me to call PB_8 PC8 as in example, maybe this is the problem

**Reply by davetcc** [reply]
It's best to start with the simplest of programs/sketches rather than try to start with a full menu application. What I would suggest, is firstly, that you use some of the included Arduino IDE sketches on your board. Get a feel for how the input and output works, check that you're understanding of pin definitions and available pins is right.

IIRC StmDuino has two sets of pin definitions, one that is native hardware level pin definitions (IE from the STM provided HAL), and one that is Arduino pin definitions, check carefully which you are using, if in doubt, maybe raise on the stm32duino forum for guidance there.

**Reply by davetcc** [reply]
I suspect you need to use the pin versions from Arduino which DON'T have an underscore in their name, as an example this is what I pulled from the TcMenu example for Stm32Duino:

> menuMgr.initForEncoder(&renderer, &menu33, PC8, PC10, PC9);

Always worth consulting the examples, there's usually at least one example for the most common board types. In this case `arduino32/stm32DuinoDemo`

**Reply by davetcc** [reply]
To be honest, when I first used Stm32Duino, this same issue hit me and it took me quite a few hours to work out what was wrong.
